pkg/providers/vmware/client/client.go

```diff
@@ -0,0 +1,173 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/session"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/mo"
+	"github.com/vmware/govmomi/vim25/soap"
+	"github.com/vmware/govmomi/vim25/types"
+	"net/url"
+	"time"
+)
+
+// timeout value in seconds for vmware api requests
+const timeout = 30 * time.Second
+
+// RichVmwareClient is responsible for retrieving VM data from the VMware API.
+type RichVmwareClient struct {
+	client         *vim25.Client
+	user           *url.Userinfo
+	sessionManager *session.Manager
+}
+
+// NewRichVMwareClient creates a new, connected rich VMWare client.
```

Comment:
nit: NewRichVMwareClient ->NewRichVMWareClient in the comment or the other way round with the function name; the linter complains if the first word in the comment on an exported function is not spelled exactly like that function's name.